codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameLogger.java

```diff
@@ -61,74 +61,106 @@ private Http2FrameLogger(InternalLogLevel level, InternalLogger logger) {
+        }
     }
 
     public void logSettings(Direction direction, ChannelHandlerContext ctx, Http2Settings settings) {
-        logger.log(level, "{} {} SETTINGS: ack=false settings={}", ctx.channel(), direction.name(), settings);
+        if (isEnable()) {
+            logger.log(level, "{} {} SETTINGS: ack=false settings={}", ctx.channel(), direction.name(), settings);
+        }
     }
 
     public void logPing(Direction direction, ChannelHandlerContext ctx, ByteBuf data) {
-        logger.log(level, "{} {} PING: ack=false length={} bytes={}", ctx.channel(),
-                direction.name(), data.readableBytes(), toString(data));
+        if (isEnable()) {
+            logger.log(level, "{} {} PING: ack=false length={} bytes={}", ctx.channel(),
+                    direction.name(), data.readableBytes(), toString(data));
+        }
     }
 
     public void logPingAck(Direction direction, ChannelHandlerContext ctx, ByteBuf data) {
-        logger.log(level, "{} {} PING: ack=true length={} bytes={}", ctx.channel(),
-                direction.name(), data.readableBytes(), toString(data));
+        if (isEnable()) {
+            logger.log(level, "{} {} PING: ack=true length={} bytes={}", ctx.channel(),
+                    direction.name(), data.readableBytes(), toString(data));
+        }
     }
 
     public void logPushPromise(Direction direction, ChannelHandlerContext ctx, int streamId, int promisedStreamId,
             Http2Headers headers, int padding) {
-        logger.log(level, "{} {} PUSH_PROMISE: streamId={} promisedStreamId={} headers={} padding={}", ctx.channel(),
-                direction.name(), streamId, promisedStreamId, headers, padding);
+        if (isEnable()) {
+            logger.log(level, "{} {} PUSH_PROMISE: streamId={} promisedStreamId={} headers={} padding={}",
+                    ctx.channel(), direction.name(), streamId, promisedStreamId, headers, padding);
+        }
     }
 
     public void logGoAway(Direction direction, ChannelHandlerContext ctx, int lastStreamId, long errorCode,
             ByteBuf debugData) {
-        logger.log(level, "{} {} GO_AWAY: lastStreamId={} errorCode={} length={} bytes={}", ctx.channel(),
-                direction.name(), lastStreamId, errorCode, debugData.readableBytes(), toString(debugData));
+        if (isEnable()) {
+            logger.log(level, "{} {} GO_AWAY: lastStreamId={} errorCode={} length={} bytes={}", ctx.channel(),
+                    direction.name(), lastStreamId, errorCode, debugData.readableBytes(), toString(debugData));
+        }
     }
 
     public void logWindowsUpdate(Direction direction, ChannelHandlerContext ctx, int streamId,
             int windowSizeIncrement) {
-        logger.log(level, "{} {} WINDOW_UPDATE: streamId={} windowSizeIncrement={}", ctx.channel(),
-                direction.name(), streamId, windowSizeIncrement);
+        if (isEnable()) {
```

Comment:
The log level will be enforced by the logger. We should only have to "double check" the log level if creation of one of the arguments may potentially be expensive. We should only need this double check on methods with a `toString(..)` as a parameter.

Alternatively we could isolate the check in `toString`:

```java
private String toString(ByteBuf buf) {
        if (!logger.isEnabled(level)) {
           return "";
        }
        if (level == InternalLogLevel.TRACE || buf.readableBytes() <= BUFFER_LENGTH_THRESHOLD) {
            // Log the entire buffer.
            return ByteBufUtil.hexDump(buf);
        }

        // Otherwise just log the first 64 bytes.
        int length = Math.min(buf.readableBytes(), BUFFER_LENGTH_THRESHOLD);
        return ByteBufUtil.hexDump(buf, buf.readerIndex(), length) + "...";
    }
```